File: ruby/cross_context_adapter/lib/gameplay_game_initializer.rb
require 'organizing/game_id'
require 'organizing/game_initializer'
require 'gameplay/board_id'
require 'gameplay/setup_board'

class GameplayGameInitializer
  include Organizing::GameInitializer

  def initialize(setup_board)
    @setup_board = setup_board
  end

  def initialize_game(white_player, black_player)
    board_id = @setup_board.call(white_player.name, black_player.name)
    Organizing::GameId.new(value: board_id.value)
  end
end

File: python/cross_context_adapter/gameplay_game_initializer.py
from gameplay_policy.board_id import BoardId
from organizing_policy.game_id import GameId
from organizing_policy.game_initializer import GameInitializer
from organizing_policy.player import Player


class GameplayGameInitializer(GameInitializer):
    def __init__(self, setup_board):
        self._setup_board = setup_board

    def initialize_game(self, white: Player, black: Player) -> GameId:
        board_id = self._setup_board(white.name, black.name)
        return GameId(value=board_id.value)
